From cd3954678738bc0dc0825824293f3b428bd1f021 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Date: Fri, 7 Mar 2025 11:01:07 +0100
Subject: [PATCH 1669/1695] dt-bindings: add MCE encryption in reserved
 memories description

Add RIF_ENC_MCE_EN in encryption values (includes side-channel
attack protection).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8c0640eb008a2792ca860fbb4a798778ceebc973 dt-bindings: add MCE encryption in reserved memories description]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Change-Id: Ifda671891f9e906c7743f772a9a8c330e12fb6c8
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/443822
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 .../devicetree/bindings/reserved-memory/reserved-memory.yaml     | 1 +
 1 file changed, 1 insertion(+)

diff --git a/documentation/devicetree/bindings/reserved-memory/reserved-memory.yaml b/documentation/devicetree/bindings/reserved-memory/reserved-memory.yaml
index 080d9ad09..d759b9676 100644
--- a/documentation/devicetree/bindings/reserved-memory/reserved-memory.yaml
+++ b/documentation/devicetree/bindings/reserved-memory/reserved-memory.yaml
@@ -133,6 +133,7 @@ properties:
 
         "enc" States that the region is encrypted.
           RIF_ENC_DIS, encryption is disabled
+          RIF_ENC_MCE_EN, encryption is enabled (side-channel attack protection)
           RIF_ENC_EN, encryption is enabled
 
         "enabled" Enables the region. Access controls applies.
-- 
2.47.2

